lib/preview-web/src/StoryRender.ts

```diff
@@ -199,6 +195,14 @@ export class StoryRender<TFramework extends AnyFramework> implements Render<TFra
         name,
         story: name,
         ...this.callbacks,
+        showError: (error) => {
+          this.phase = 'errored';
```

Comment:
Do we need to emit the phase changed event? Not that we use it anywhere, but we should probably be consistent.

Reply:
That's already done by `PreviewWeb.renderException` which is invoked through the `showException` callback. Arguably a `renderX` function shouldn't be emitting events as a side-effect though. But I don't want to extend the scope of this PR any further.